### DISCUSSION TITLE
When using export_code() to convert casadi function to matlab function, how to avoid the sparse matrix?

### DISCUSSION THREAD

**Discussion by zhangjiadi44** [casadi/casadi#3447] [Q&A]
As described in the title, how to make the output a full matrix instead of a sparse matrix? Is there any option in export_code() function?

**Comment by jaeandersson** [comment]
Try to compile the MEX file with while defining the variable `CASADI_MEX_NO_SPARSE` during compilation

**Reply by jaeandersson** [reply]
Alternatively, make sure that the output expression has no structural zeros before generating code. For example by using `densify`. Sketch:
```matlab
J = jacobian(x, f);
J = densify(J);  % remove any structural nonzeros
F = casadi.Function('F', {x}, {J}, {'x'}, {'J'})
F.generate('F.c', struct('mex', true));
```
